File: pomodoro_app/lib/models/task.dart
class Task {

  final String id;
  final String title;

  int totalPomodoro;
  int completedPomodoro;

  bool completed;
  Task({
    required this.id,
    required this.title,
    required this.totalPomodoro,
    required this.completedPomodoro,
    required this.completed,
  });

  factory Task.fromJson(Map<String, dynamic> json) {

    return Task(
      id: json["_id"],
      title: json["title"],
      totalPomodoro: json["totalPomodoro"] ?? 1,
      completedPomodoro: json["completedPomodoro"] ?? 0,
      completed: json["completed"] ?? false,
    );

  }
}

File: backend/models/Task.js
const mongoose = require("mongoose");

const taskSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "User"
  },

  title: {
    type: String,
    required: true
  },

  totalPomodoro: {
    type: Number,
    default: 1
  },

  completedPomodoro: {
    type: Number,
    default: 0
  },

  completed: {
    type: Boolean,
    default: false
  }

});

module.exports = mongoose.model("Task", taskSchema);